src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,61 @@
+package duke;
+
+import java.util.Arrays;
+
+/**
+ * Encapsulates parsing of logic.
+ */
+public class Parser {
+    /**
+     * Returns appropriate Command given user input.
+     * @param fullCommand User input.
+     * @return Command of user input.
+     * @throws DukeException If user inputs invalid command.
+     */
+    public static Command parse(String fullCommand) throws DukeException {
+        String[] strings = fullCommand.split(" ");
+        String description = String.join(" ", Arrays.copyOfRange(strings, 1, strings.length));
+        switch (convertToEnum(strings[0])) {
+        case LIST:
+            if (strings.length > 1) {
+                throw new DukeException("Please key in a correct command.");
+            }
+            return new ShowCommand();
+        case DONE:
+            return new UpdateCommand(description);
+        case DELETE:
+            return new DeleteCommand(description);
+        case FIND:
+            return new FindCommand(description);
+        case TODO: case EVENT: case DEADLINE:
+            return new AddCommand(convertToEnum(strings[0]), description);
+        case BYE:
+            return new ExitCommand();
+        default:
+            throw new DukeException("Please key in a correct command.");
+        }
+    }
+
+    private static CommandEnum convertToEnum(String string) throws DukeException {
```

Comment:
Might be a little nitpicking, but should the method name be more detailed (e.g. convertCommandToEnum)? (See Week 4 -> Topics -> W4.6d)

Reply:
Indeed, i should have renamed the method name to make it more intuitive for readers. Thanks for the comment!